### PR TITLE
kafka-ingest: small ergonomic cleanups.

### PR DESCRIPTION
**Pull request by ruchirK**
### Motivation

    This commit makes the open loop benchmark slightly easier to use by:
    - adding an argument to vary the number of timely workers
    - adding help strings for all arguments
    - forbidding the use of --s3-storage without --enable-persistence

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: test/kafka-ingest-open-loop/mzcompose.py [hunk 1]
```diff
@@ -119,6 +119,12 @@ def workflow_kafka_ingest_open_loop(
         action="store_true",
         help="whether or not to enable persistence on materialized",
     )
+    parser.add_argument(
+        "--workers",
+        type=int,
+        default=4,
```
Comment (danhhz) [inline test/kafka-ingest-open-loop/mzcompose.py:127]: I think it works if we default to None instead. following down into the Materialized definition, looks like a None omits the --workers flag and we get the default (num cpus), which is my preference for the behavior here

**Review by danhhz — APPROVED** [review]
wow that's all we need to get it to work with our dev prometheus/grafana?

**Comment by ruchirK** [comment]
> wow that's all we need to get it to work with our dev prometheus/grafana?

Yeah that's what I thought from looking at the existing chbench and perf-kinesis tests which don't do anything different. Perhaps unsurprisingly though, its not actually working on an ec2 instance I thought the fix might be to add 

```
+    "tags": {
+        "scrape_benchmark_numbers": "true",
+    }
```

to the machine config but that didn't seem to work either. This will be something to investigate tomorrow!

File: test/kafka-ingest-open-loop/mzcompose.py (hunk 1 above)
Comment (ruchirK) [reply test/kafka-ingest-open-loop/mzcompose.py]: Done

**Comment by ruchirK** [comment]
Learned last week that grafana works out of the box on scratch as long as we specify `--preserve-ports` when running mzcompose and run with `scrape_benchmark_numbers: true`. I'm going to compile all of those benchmark changes separately, and figured all 3 of these cleanups are worth having as well. TFTR + merging on green!

**Pull request merged**